src/Testing/MockUnaryCall.php

```diff
@@ -72,7 +74,9 @@ public function __construct($response, $deserialize = null, $status = null)
      */
     public function wait()
     {
-        $obj = $this->deserializeMessage($this->response, $this->deserialize);
-        return [$obj, $this->status];
+        if ($this->status->code != \Google\Rpc\Code::OK) {
```

Comment:
This change makes sense so long as this class is going to be used to mock the response from the TransportInterface. If it is going to be used to mock the response from the gRPC stub (which is how it is used currently) then I don't think you need this change.